ethers-solc/src/compile.rs

```diff
@@ -78,7 +83,17 @@ pub struct Solc {
 
 impl Default for Solc {
     fn default() -> Self {
-        std::env::var("SOLC_PATH").map(Solc::new).unwrap_or_else(|_| Solc::new(SOLC))
+        if let Ok(solc) = std::env::var("SOLC_PATH") {
+            return Solc::new(solc)
+        }
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            if let Some(solc) = Solc::svm_global_version().filter(|p| p.exists()) {
+                return Solc::new(solc)
```

Comment:
.global-version returns only the version, not the path to the binary with that version. So we need an extra resolution from version to path

Reply:
oh you're right, somehow I thought this was a copy